From 63a14ba4e10dd38f223f3e0bc14eda295591d6a6 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 22 Nov 2024 08:45:07 +0100
Subject: [PATCH 1295/1695] scripts: sign_rproc_fw: fix cryptography import

On windows PC following error is reported running the script:

from cryptography.hazmat.primitives.ciphers.aead import utils
ImportError: cannot import name 'utils' from
'cryptography.hazmat.primitives.ciphers.aead'

Fix the import to be compliant with windows PC.

Fixes: 41190c54a0f5 ("scripts: sign_rproc_fw: Add support for AES CBC encryption")

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 401510eb97f2e655e59407a961ba3cd7d20f920b scripts: sign_rproc_fw: fix cryptography import]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I1bc583191abd755b554f4a8a3080ec1d360a3a85
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/420224
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 scripts/sign_rproc_fw.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/sign_rproc_fw.py b/scripts/sign_rproc_fw.py
index c70155355..75a596bc2 100755
--- a/scripts/sign_rproc_fw.py
+++ b/scripts/sign_rproc_fw.py
@@ -24,7 +24,7 @@ try:
     from Cryptodome.Signature import DSS
     from Cryptodome.PublicKey import ECC
     from Cryptodome.Cipher import AES
-    from cryptography.hazmat.primitives.ciphers.aead import utils
+    from cryptography import utils
     from Cryptodome.Random import get_random_bytes
 except ImportError:
     print("""
-- 
2.47.2

